src/utils/next.ts

```diff
@@ -59,6 +60,17 @@ const hasProps = (result: any): result is ResultWithProps => {
     return true;
 };
 
+const stripParams = (relativePath: string, params?: ParsedUrlQuery) => {
```

Comment:
To make this function more general-purpose and more clearly communicate what it does, I would have made the second argument a `string[]` and moved the use of `Object.keys()` outside of the function to the call site, so that the function is called like so:

```tsx
const attemptedPath = stripParams(ctx.resolvedUrl, Object.keys(ctx.params));
```

But ultimately that's a matter of taste and not something that needs to change. AHA! 😊